src/mode.js

```diff
@@ -43,59 +40,56 @@ let fullVersion = '';
 
 /**
  * Provides info about the current app.
+ * @param {!Window=} win
  * @return {!ModeDef}
  */
-export function getMode() {
-  if (mode) {
-    return mode;
+export function getMode(opt_win) {
```

Comment:
Will this have any effect on us being able to DCE this code?

/cc @erwinmombay